docs/aws/audit/iamcompliance/rules/iam_customer_policies_for_blocked_kms_actions_remediation.mdx

```diff
@@ -1,6 +1,283 @@
+           for statement in policy_document['Statement']:
+               if 'kms:' in statement['Action']:
+                   print(f"KMS actions are correctly set in policy: {policy_arn}")
+               else:
+                   print(f"KMS actions are missing in policy: {policy_arn}")
+
+   validate_policies()
+   ```
+
+This script will help you ensure that KMS actions are properly set in IAM policies by retrieving existing policies, updating them if necessary, and validating the changes.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the IAM dashboard by selecting "Services" from the top menu, then choosing "IAM" under the "Security, Identity, & Compliance" category.
+3. In the IAM dashboard, select "Policies" from the left-hand side menu. This will display a list of all the IAM policies currently in use.
+4. For each policy, click on the policy name to open its details page. Here, you can review the policy's JSON document under the "Permissions" tab. Look for any KMS-related actions (like "kms:Encrypt", "kms:Decrypt", "kms:ReEncrypt*", etc.) that are explicitly denied. If such actions are found, it indicates that blocked KMS actions are set in the IAM policy.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access IAM policies.
+
+2. Once AWS CLI is set up, you can list all the IAM policies using the following command:
+   ```
+   aws iam list-policies --scope Local
+   ```
+   This command will return a list of all IAM policies in your AWS account.
+
+3. To check the details of each policy, you can use the following command:
+   ```
+   aws iam get-policy-version --policy-arn <Policy_ARN> --version-id <Policy_Version_ID>
+   ```
+   Replace `<Policy_ARN>` and `<Policy_Version_ID>` with the ARN and version ID of the policy you want to check. This command will return the details of the specified policy version.
+
+4. Now, you need to manually inspect the policy details to check if there are any blocked KMS actions. Look for the "Effect": "Deny" statement in the policy document. If there are any KMS actions listed under this statement, it means those actions are blocked.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Setup AWS SDK for Python (Boto3):**
+   First, you need to set up AWS SDK for Python (Boto3) in your local environment. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   After installing boto3, you need to configure it. You can do this by creating a new session using your AWS credentials:
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>